include/re_udp.h

```diff
@@ -36,7 +36,7 @@ void udp_handler_set(struct udp_sock *us, udp_recv_h *rh, void *arg);
 void udp_error_handler_set(struct udp_sock *us, udp_error_h *eh);
 int  udp_thread_attach(struct udp_sock *us);
 void udp_thread_detach(struct udp_sock *us);
-int  udp_sock_fd(const struct udp_sock *us, int af);
+re_sock_t udp_sock_fd(const struct udp_sock *us, int af);
```

Comment:
I think this function is unused, we should consider removing it.

Exposing implementation details in the public api is considered bad practice.
this detail should be hidden inside the module.